### PR TITLE
Search: Fix Search returning results out of order

### PR DESCRIPTION
**Pull request by joshhunt**
SearchStateManager will set the search results for the last query returned. If the search API responds "out of order", then this would show search results out of order. This is particularly noticeable when searching instance with many results, depending on how fast you type. For example, if trying to search for `debugging`:

 - You type the `d` and wait just a _little_ bit (300ms), we `GET search?query=d`
 - You finish typing `debugger, we `GET search?query=debugger`
 - `GET search?query=debugger` returns first because it has less results, and we render these in UI
 - `GET search?query=d` finally returns, and we _these_ results, losing the results issued last

This is especially annoying as because the first search typically has more results, it's more likely to return _later_. This has always been a subtle problem, but I believe nestedFolders has a performance impact that makes this more apparent.

This PR changes SearchStateManager to keep the previous promises, and wait until it's resolved until resolving the _new_ search promise.

### CODE REVIEW

File: public/app/features/search/state/SearchStateManager.test.ts [hunk 1]
```diff
@@ -15,6 +18,8 @@ jest.mock('@grafana/runtime', () => {
 });
 
 describe('SearchStateManager', () => {
+  // jest.spyOn(lodash, 'debounce').mock;
```
Comment (L-M-K-B) [inline public/app/features/search/state/SearchStateManager.test.ts:21]: Why not deleting this instead of keeping it commented out?

Comment (joshhunt) [reply public/app/features/search/state/SearchStateManager.test.ts:21]: sorry. this shouldn't have been committed at all 😅   
I'm not sure how this snuck in..........

**Comment by joshhunt** [comment]
Sorry for lack of clarity @tskarhed - this code change only impacts Browse Dashboards search. Not command palette.

**Comment by tskarhed** [comment]
But yes, this fix works. Just add some tests and it is all good!

**Review by tskarhed — APPROVED** [review]

**Pull request merged**